File: Python/MaximumIceCreamBars.py
class Solution(object):
    def maxIceCream(self, costs, coins):
        count = [0] * 100001

        for cost in costs:
            count[cost] += 1

        ans = 0

        for i in range(1, 100001):
            while count[i] > 0 and coins >= i:
                ans += 1
                count[i] -= 1
                coins -= i

        return ans

File: C++/MaximumIceCreamBars.cpp
class MaximumIceCreamBars {
public:
    int maxIceCream(vector<int>& costs, int coins) {
        vector<int> count(100001, 0);

        for (int cost : costs) {
            count[cost]++;
        }

        int ans = 0;

        for (int i = 1; i < count.size(); i++) {
            while (count[i] > 0 && coins >= i) {
                ans++;
                count[i]--;
                coins -= i;
            }
        }

        return ans;
    }
};